sstable/reader_virtual_test.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2023 The LevelDB-Go and Pebble Authors. All rights reserved. Use
+// of this source code is governed by a BSD-style license that can be found in
+// the LICENSE file.
+
+package sstable
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"testing"
+
+	"github.com/cockroachdb/pebble/internal/base"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPrefixReplacingIterator(t *testing.T) {
+	for _, tc := range []struct{ from, to []byte }{
+		{from: []byte("aa"), to: []byte("bb")},
+		{from: []byte("bb"), to: []byte("aa")},
+		{from: []byte("aa"), to: []byte("zzz")},
+		{from: []byte("zzz"), to: []byte("aa")},
+	} {
+		t.Run(fmt.Sprintf("%s_%s", tc.from, tc.to), func(t *testing.T) {
+			r := buildTestTable(t, 20, 256, 256, DefaultCompression, tc.from)
+			defer r.Close()
+			rawIter, err := r.NewIter(nil, nil)
+			require.NoError(t, err)
+			defer rawIter.Close()
+
+			raw := rawIter.(*singleLevelIterator)
```

Comment:
do we need it to be a singleLevelIterator? I don't see why. Might be a good idea to have a version of this test that calls `buildTestTable` with tiny index/data block sizes so we get a twoLevelIterator here too.

Reply:
we don't; the cast is so I can assert the internal fields are indeed being set by Set*.

Reply:
> do we need it to be a singleLevelIterator

No, I'm just casting it so that I can verify Set* functions set the underlying fields since otherwise their effects are unobservable.